proto/src/determined/trial/v1/trial.proto

```diff
@@ -32,23 +25,15 @@ message CheckpointWorkload {
   // Dictionary of file paths to file sizes in bytes of all files in the
   // checkpoint.
   map<string, int64> resources = 5;
-  // Number of batches in this workload.
-  int32 num_batches = 6;
-  // Number of batches processed prior to this workload.
-  int32 prior_batches_processed = 7;
+  // Total number of batches as of this workload's completion.
+  int32 total_batches = 6;
```

Comment:
This is technically a breaking change, though I think the only alternative is deprecating it and then breaking it next release, not this one.

Reply:
@hamidzr wdyt?

Reply:
hmm we haven't set up a deprecation or API versioning strategy yet to follow in these situations. In lieu of those, we should make sure we put the changes in our release notes as breaking changes and use the API change GitHub label. In terms of our own usage, we migrate relatively easily but we don't really have a way of telling what our external users are doing with the APIs..  Since we still have that beta tag in our APIs I think we'd be okay to just remove. If we're not looking to immediately remove support for these fields that we are removing from the system leaving them exposed as deprecated for a release and removing them later doesn't seem viable.

Reply:
also ditto on reusing field numbers